### ISSUE TITLE
Project registration and service lifetime

### ISSUE THREAD

**Issue by VelvetThePanda** [martinothamar/Mediator#58]
This library is neat! I quite enjoy it and the concept, and when it works, it *works*.

However, I've run into a peculiar issue, in a similar vein of #51; I have two projects (at least, for the sake of this issue): `Application` and `Data`. The latter of the two houses requests and handlers (I should break this out, but stick with me), and I initially installed the source generator on that project, ran into some issues, followed the steps outlined in the aforementioned issue, and now I'm even worse off.

Rider doesn't see the extension method (and in fact, I don't either), but the project technically builds fine; furthermore, I can't seem to set the service registration correctly (though, this could/may be its own issue entirely).

(Also: I didn't want to necro the issue mentioned, and this is a smidge more complicated than that from what I can gather, hence the new issue.)

**Comment by VelvetThePanda** [comment]
In short:

```cs
// Application project
// AddMediator isn't recognized, nor is the generator run, which is peculiar.
services.AddMediator(options => options.ServiceLifetime = ServiceLifetime.Scoped);
```

The csproj for that project:
```xml
<ItemGroup>
  <PackageReference Include="Mediator.Abstractions" Version="2.1.0-preview.2" />
  <PackageReference Include="Mediator.SourceGenerator" Version="2.1.0-preview.2">
    <PrivateAssets>all</PrivateAssets>
    <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
  </PackageReference>
</ItemGroup>

<ItemGroup>
  <ProjectReference Include="..\Silk.Data\Silk.Data.csproj" />
</ItemGroup>
```

And for the data project:

```xml
<ItemGroup>
  <PackageReference Include="Mediator.Abstractions" Version="2.1.0-preview.2" />
</ItemGroup>
```

**Comment by VelvetThePanda** [comment]
Really appreciate the quick response! Yes, that's about right in terms of structure, which only makes this all the more perplexing. 
![image](https://user-images.githubusercontent.com/42438262/202794461-372e13af-b883-4765-adff-c22bfe0fde56.png)

**Comment by VelvetThePanda** [comment]
using `dotnet build` runs fine, but Roslyn appears to be performing voodoo magic, because there's no generated file in any of the common locations (and the service registration doesn't appear to be working, given the runtime errors from EF Core)

**Comment by martinothamar** [comment]
I tried to make it .NET 7 and installed Rider, now I think I see the issue 
![image](https://user-images.githubusercontent.com/5425986/202795960-e8cf7554-5caf-43c9-bf04-95f797e5a827.png)

This application still runs also in Rider, but syntax highlighting seems to be bugged out, and there is the warning box about not supporting my installed .NET 7 SDK

One tip I have on seeing the generated code - add this attribute to the Application.csproj `PropertyGroup`:
```
    <EmitCompilerGeneratedFiles>true</EmitCompilerGeneratedFiles> 
```
Then you should be able to inspect the generated code in the path depicted below
![image](https://user-images.githubusercontent.com/5425986/202796355-264e038e-59c3-44b7-9c59-a84040064f62.png)

That way you'll know if there are any obvious issues, but generally if it builds it should be OK, atleast thats my experience

**Comment by VelvetThePanda** [comment]
Hm. I use the beta versions personally, so .NET 7 is mostly supported there. If I install it in the data project, it works fine (as in, the syntax is highlighted, but same runtime issues), of which seems to be addressed in #50. I do appreciate the csproj tip, though!
